File: leetcode/Two_Sum.java
import java.util.*;

public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < numbers.length; ++i)
            map.put(numbers[i], i + 1);
        int[] ret = new int[2];
        for (int i = 0; i < numbers.length; ++i) {
            int x = target - numbers[i];
            if (map.containsKey(x)) {
                ret[0] = i + 1;
                ret[1] = map.get(x);
                break;
            }
        }
        return ret;
    }
}

File: leetcode/Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        map<int, int> has;
        vector<int> indices;
        for (int i = 0; i < numbers.size(); ++i) {
            int rest = target - numbers[i];
            if (has.find(rest) != has.end()) {
                indices.push_back(has[rest] + 1);
                indices.push_back(i + 1);
                return indices;
            } else {
                has[numbers[i]] = i;
            }
        }
    }
};
